<?xml version="1.0" encoding="UTF-8" ?>
<robot name="sia10">
    <link name="base_link">
        <visual>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/visual/base_link.stl" />
            </geometry>
            <material name="link_color_bl">
                <color rgba="0.75 0.75 0.75 1" />
            </material>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/collision/base_link_c.stl" />
            </geometry>
            <material name="collision_color_bl">
                <color rgba="0 1 1 1" />
            </material>
        </collision>
    </link>
    <link name="link_s">
        <visual>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/visual/link_s.stl" />
            </geometry>
            <material name="link_color_ls">
                <color rgba="0.75 0.75 0.75 1" />
            </material>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/collision/link_s_c.stl" />
            </geometry>
            <material name="collision_color_ls">
                <color rgba="0 1 1 1" />
            </material>
        </collision>
    </link>
    <link name="link_l">
        <visual>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/visual/link_l.stl" />
            </geometry>
            <material name="link_color_ll">
                <color rgba="0 0 1 1" />
            </material>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/collision/link_l_c.stl" />
            </geometry>
            <material name="collision_color_ll">
                <color rgba="0 1 1 1" />
            </material>
        </collision>
    </link>
    <link name="link_e">
        <visual>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/visual/link_e.stl" />
            </geometry>
            <material name="link_color_le">
                <color rgba="0 0 1 1" />
            </material>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/collision/link_e_c.stl" />
            </geometry>
            <material name="collision_color_le">
                <color rgba="0 1 1 1" />
            </material>
        </collision>
    </link>
    <link name="link_u">
        <visual>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/visual/link_u.stl" />
            </geometry>
            <material name="link_color_lu">
                <color rgba="0.75 0.75 0.75 1" />
            </material>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/collision/link_u_c.stl" />
            </geometry>
            <material name="collision_color_lu">
                <color rgba="0 1 1 1" />
            </material>
        </collision>
    </link>
    <link name="link_r">
        <visual>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/visual/link_r.stl" />
            </geometry>
            <material name="link_color_lr">
                <color rgba="0.75 0.75 0.75 1" />
            </material>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/collision/link_r_c.stl" />
            </geometry>
            <material name="collision_color_lr">
                <color rgba="0 1 1 1" />
            </material>
        </collision>
    </link>
    <link name="link_b">
        <visual>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/visual/link_b.stl" />
            </geometry>
            <material name="link_color_lb">
                <color rgba="0 0 1 1" />
            </material>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/collision/link_b_c.stl" />
            </geometry>
            <material name="collision_color_lb">
                <color rgba="0 1 1 1" />
            </material>
        </collision>
    </link>
    <link name="link_t">
        <visual>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/visual/link_t.stl" />
            </geometry>
            <material name="link_color_lt">
                <color rgba="0.67 0.57 0.19 1" />
            </material>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://sia10_simple_description/meshes/collision/link_t_c.stl" />
            </geometry>
            <material name="link_color">
                <color rgba="0 1 1 1" />
            </material>
        </collision>
    </link>


    <joint name="joint_s" type="revolute">
        <parent link="base_link" />
        <child link="link_s" />
        <origin xyz="0 0 0.18" rpy="0 0 0" />
        <axis xyz="0 0 1" />
        <limit effort="0" lower="-3.13" upper="3.13" velocity="2.95" />
    </joint>
    <joint name="joint_l" type="revolute">
        <parent link="link_s" />
        <child link="link_l" />
        <origin xyz="0 0 0.18" rpy="0 0 0" />
        <axis xyz="1 0 0" />
        <limit effort="0" lower="-1.90" upper="1.9" velocity="2.95" />
    </joint>
    <joint name="joint_e" type="revolute">
        <parent link="link_l" />
        <child link="link_e" />
        <origin xyz="0 0 0.18" rpy="0 0 0" />
        <axis xyz="0 0 1" />
        <limit effort="0" lower="-2.95" upper="2.95" velocity="2.95" />
    </joint>
    <joint name="joint_u" type="revolute">
        <parent link="link_e" />
        <child link="link_u" />
        <origin xyz="0 0 0.18" rpy="0 0 0" />
        <axis xyz="-1 0 0" />
        <limit effort="0" lower="-2.36" upper="2.36" velocity="2.95" />
    </joint>
    <joint name="joint_r" type="revolute">
        <parent link="link_u" />
        <child link="link_r" />
        <origin xyz="0 0 0.18" rpy="0 0 0" />
        <axis xyz="0 0 -1" />
        <limit effort="0" lower="-3.13" upper="3.13" velocity="3.48" />
    </joint>
    <joint name="joint_b" type="revolute">
        <parent link="link_r" />
        <child link="link_b" />
        <origin xyz="0 0 0.18" rpy="0 0 0" />
        <axis xyz="-1 0 0" />
        <limit effort="0" lower="-1.90" upper="1.9" velocity="3.48" />
    </joint>
    <joint name="joint_t" type="revolute">
        <parent link="link_b" />
        <child link="link_t" />
        <origin xyz="0 0 0.155" rpy="0 0 0" />
        <axis xyz="0 0 -1" />
        <limit effort="0" lower="-3.14" upper="3.14" velocity="6.97" />
    </joint>
</robot>
